code/modules/antagonists/changeling/powers/mutations.dm

```diff
@@ -187,7 +187,8 @@
 	else if(istype(target, /obj/machinery/door/airlock))
 		var/obj/machinery/door/airlock/A = target
 
-		if((!A.requiresID() || A.allowed(user)) && A.hasPower()) //This is to prevent stupid shit like hitting a door with an arm blade, the door opening because you have acces and still getting a "the airlocks motors resist our efforts to force it" message, power requirement is so this doesn't stop unpowered doors from being pried open if you have access
+		if((A.id_scan_hacked() || A.allowed(user)) && A.hasPower()) //This is to prevent stupid shit like hitting a door with an arm blade, the door opening because you have access and still getting a "the airlocks motors resist our efforts to force it" message, power requirement is so this doesn't stop unpowered doors from being pried open if you have access.
+			//Note that because the id_scan_hacked() check is the opposite from how it works in the actual opening check, a powered + id_scan_hacked airlock will prevent lings from forcing the door with arm blades, while also blocking regular access. Also, the entire premise of the comment/logic above is flawed, since you can only use arm blades on a door with HARM intent (due to this being afterattack()), and that prevents you from interacting normally.
```

Comment:
Changelings attacking the door on any intent should just pry the door rather than perform ID checks. I don't get why removing ID access would make it so changelings can't pry open a door

Reply:
I think this is actually several questions, I'll try to answer them all.
- Changelings attacking the door on any intent should just pry, why don't they?
  - This is because the pry functionality is defined in `afterattack()`, and `afterattack()` only gets called if `attackby()` returns FALSE. (See `melee_attack_chain` in `_onclick/item_attack.dm`.) Door's `attackby()` _unconditionally_ returns 1 if the attacking item is BLUDGEON and intent isn't harm, thus bypassing the `afterattack()`.
- Why does removing ID access make it so they can't pry the door?
  - This one is pretty clear from the code, but I suspect you weren't asking literally...
- Why _should_ removing ID access make it so they can't pry the door?
  - Almost certainly, it shouldn't. It's a bug based on misunderstanding how (formerly) `requiresID()` worked, because of how this stuff was named and documented.
- Why haven't I done something about it?
  - I tried to fix what I thought was a bug before, but it turned out to be an important feature. :) In this change, I strove to make no behavior changes at all, and just document things that were weird. If you think I should change something here, I'm happy too, but there are several possible options of what/how much to change. (Leave arm blades only working on HARM intent and rip out this whole clause? Just take out the part relating to id scan? Try to make arm blades work on all intents?)